### PR TITLE
Fix/cs 326 check existence of proof purpose

### PR DESCRIPTION
**Pull request by PManaras**

### CODE REVIEW

File: src/main/java/org/eclipse/tractusx/ssi/lib/model/proof/Proof.java
Comment (mustafasalfiti) [inline src/main/java/org/eclipse/tractusx/ssi/lib/model/proof/Proof.java:1]: my question: should each proof contain a proofPurpose ? or this is something depended on the Type of the proof ? in that case a new class of proof should be created and should inherit from this

**Review by mustafasalfiti — APPROVED** [review]

File: src/main/java/org/eclipse/tractusx/ssi/lib/model/proof/Proof.java
Comment (hkny) [reply src/main/java/org/eclipse/tractusx/ssi/lib/model/proof/Proof.java:1]: Assertion for VCs and Authentication for VPs should be the proof purposes of all VCs/VPs

**Comment by mustafasalfiti** [comment]
@hkny can we get an update on that ? we discussed this before that not every proof have proofPurpose but in our case in MIW every proof should have a proofPurpose. 
should we merge that or just close the PR ?

**Comment by mustafasalfiti** [comment]
implemented feature

**Pull request closed without merging**